src/toil/batchSystems/slurm.py

```diff
@@ -444,6 +443,29 @@ def prepareSbatch(self,
             if cpu is not None:
                 sbatch_line.append(f'--cpus-per-task={math.ceil(cpu)}')
 
+            if gpus:
+                # This block will add a gpu supported partition only if no partition is supplied by the user
+                sbatch_line = sbatch_line[:1] + [f'--gres=gpu:{gpus}'] + sbatch_line[1:]
+                lowest_priority_gpus = sorted([i for i in self.boss.partitions if i["gres"]], key=lambda x: x["priority"])
+                new_partition_name = lowest_priority_gpus[0][0]  # get name of lowest priority partition that supports gpus
```

Comment:
This looks like it is going to cause an IndexError when the user hasn't specified a partition and a GPU job is run but no partition has GPUs, since the list is empty then. It's also going to cause an IndexError when the user *has* specified a partition, and runs a GPU job, but no partition appears to have GPUs, since we try and find a default partition whether we need one or not.

I think in the first case we want to [raise an `InsufficientSystemResources`](https://github.com/DataBiosphere/toil/blob/937d4fe5964a9f5a0faaeb5e5156c58ae9a593df/src/toil/batchSystems/abstractBatchSystem.py#L586) to complain that wee have no GPUs on Slurm, and in the second case we want to just print the warning we have about the selected partition not having GPUs.